### ISSUE TITLE
[spir64] Wrong kernels when a loop is unrolled

### ISSUE THREAD

**Issue by Michoumichmich** [intel/llvm#4527]
**Describe the bug**
If I unroll [this](https://github.com/Michoumichmich/SYCL-Hashing-Algorithms/blob/c4e21f98ccd3fa956309d05610d31a96b5a37d29/src/hash_functions/sha1.cpp#L40) loop:

```C++
    for (qword i = 0, j = 0; i < 16; ++i, j += 4)
        m[i] = (dword) (((dword)data[j] << 24) + ((dword)data[j + 1] << 16) + ((dword)data[j + 2] << 8) + ((dword)data[j + 3]));
```

DPC++ generates, in debug mode, on the spir64 target a code that behaves in an incorrect manner. There is no crash, it's just the result of the hashing function that is wrong. 

The code behaves properly, in debug and release mode, on the CUDA back-end as well as on other SYCL implementations.

No warnings are produced.

**To reproduce**
```
git clone https://github.com/Michoumichmich/SYCL-Hashing-Algorithms sycl_hash
cd sycl_hash ; mkdir build ; cd build
CXX=clang++ cmake .. -DCMAKE_BUILD_TYPE=Debug 
make all
```

**Environment (please complete the following information):**

- OS: Linux
- Target device and vendor: Intel OpenCL CPU
- DPC++ version: 9a0ea9ae3dca23f16233ba211032cf20fddd6579

**Additional context**

**Comment by Michoumichmich** [comment]
Hello @bader, sorry for replying a little bit late. Yes, the result of the hashing function is wrong only when applying `#pragma unroll` in debug and on OpenCL. 

The target to run is just the test suite with `make test`. It will run the hashing functions on all of the available devices. 

Somehow my OpenCL installation broke and I'm unable to run the code on OpenCL/CPU. I was using `intel-oneapi-runtime-opencl`. What should I use instead? or do you have a clue what changed/should I change ?
